x-pack/plugins/observability/common/custom_threshold_rule/get_view_in_app_url.test.ts

```diff
@@ -5,15 +5,15 @@
  * 2.0.
  */
 
-import { DiscoverAppLocatorParams } from '@kbn/discover-plugin/common';
 import { Aggregators } from './types';
 import { LocatorPublic } from '@kbn/share-plugin/common';
+import { LogsExplorerLocatorParams } from '@kbn/deeplinks-observability';
 import { getViewInAppUrl, GetViewInAppUrlArgs } from './get_view_in_app_url';
 
 describe('getViewInAppUrl', () => {
   const logsExplorerLocator = {
     getRedirectUrl: jest.fn(() => 'mockedGetRedirectUrl'),
-  } as unknown as LocatorPublic<DiscoverAppLocatorParams>;
+  } as unknown as LocatorPublic<LogsExplorerLocatorParams>;
```

Comment:
@maryam-saeidi LogExplorerLocatorParams define a new and more complex type for `columns` which the Discover State only defines as `string[]`

Reply:
In our case, we are providing a data view and as a result, we will land on discover in the end, isn't `DiscoverAppLocatorParams` more suitable for our use case? I am trying to understand what is the reason for this change in observability, was there an issue?

Reply:
I don't get it, its a LogExplorerLocator, which will land you on LogExplorer.
Also logExplorerLocator should use `LogsExplorerLocatorParams` instead of `DiscoverAppLocatorParams`.

The reason to change in observability is we have introduced a change in the LogExplorerLocator and hence the types need to be updated. Previously the type was almost same as Discover but now we need a more complex type.

Reply:
> I don't get it, its a LogExplorerLocator, which will land you on LogExplorer.

I mean for the data view related use case, we will send the user to Logs Explorer but then it will be redirected to Discover. I guess in that case, the Logs Explorer locator should know how to handle the transition to discover, so using `LogsExplorerLocatorParams` probably is fine.

Reply:
Yes, it is still the same locator, just changed the type to support passing different column config